### ISSUE TITLE
Storage failed, but the charge was deducted

### ISSUE THREAD

**Issue by pengxiankaikai** [filecoin-project/go-filecoin#2123]
**Describe the bug**
At the beginning
go-filecoin wallet balance fcqffgl0usd3pc76kj0mk4883ayl6stjrq690ww5r
899.999972352

go-filecoin client list-asks --enc=json | jq
{
  "Miner": "fcqaz0293zaut8npw4te9mf4f2u2xvfplmkc65xf4",
  "Price": "0.001",
  "Expiry": 28669,
  "ID": 0,
  "Error": null
}

go-filecoin client propose-storage-deal fcqaz0293zaut8npw4te9mf4f2u2xvfplmkc65xf4 Qme88CPSMQMzHZ7svjBRvq1e9dwi2GWhNSum7eJGGsweNQ 0 2880
Error: error sending proposal: failed to establish connection with the peer: context deadline exceeded

return Error, but my token decrease！

go-filecoin wallet balance fcqffgl0usd3pc76kj0mk4883ayl6stjrq690ww5r
761.759972352

thanks!

**Comment by shannonwells** [comment]
I did a little preliminary investigation.  A simulated `context deadline exceeded` error in test did not result in a deduction of FIL.  The following test asserts that the balance is the same as the initial balance after such a failure, caused by the minerDaemon . However, this test doesn't produce exactly the same error as above "failed to establish connection with the peer", so maybe the conditions set up in the test aren't right. 
```
func TestDealFailures(t *testing.T) {
	t.Parallel()
	ast := assert.New(t)

	t.Run("when context deadline is exceeded, propose-storage-deal does not charge FIL", func(t* testing.T) {
		minerDaemon := th.NewDaemon(t,
			th.WithMiner(fixtures.TestMiners[0]),
			th.KeyFile(fixtures.KeyFilePaths()[0]),
			th.DefaultAddress(fixtures.TestAddresses[0]),
			th.AutoSealInterval("1"),
		).Start()
		defer minerDaemon.ShutdownSuccess()

		clientDaemon := th.NewDaemon(t,
			th.KeyFile(fixtures.KeyFilePaths()[1]),
			th.DefaultAddress(fixtures.TestAddresses[1]),
		).Start()
		defer clientDaemon.ShutdownSuccess()
		clientDefaultAddr := clientDaemon.RunSuccess("wallet", "addrs", "ls").ReadStdoutTrimNewlines()
		initialBalance := clientDaemon.RunSuccess("wallet", "balance", clientDefaultAddr).ReadStdoutTrimNewlines()

		minerDaemon.RunSuccess("mining", "start")
		minerDaemon.UpdatePeerID()

		minerDaemon.ConnectSuccess(clientDaemon)

		minerDaemon.MinerSetPrice(fixtures.TestMiners[0], fixtures.TestAddresses[0], "20", "10")
		dataCid := clientDaemon.RunWithStdin(strings.NewReader("HODLHODLHODL"), "client", "import").ReadStdoutTrimNewlines()
		output := clientDaemon.Run("client", "propose-storage-deal", fixtures.TestMiners[0], dataCid, "0", "5").ReadStderr()
		ast.Equal("Error: error creating payment: context deadline exceeded", output)

		finalBalance := clientDaemon.Run("wallet", "balance", clientDefaultAddr).ReadStdoutTrimNewlines()
		ast.Equal(initialBalance, finalBalance)
	})
}
```

**Comment by pengxiankaikai** [comment]
@shannonwells thanks! 
At present, it seems that there has not been such a problem, and I am not sure what happened before.

**Comment by pengxiankaikai** [comment]
@shannonwells I just reappeared this problem, you can pay attention to this new problem.[#2210](https://github.com/filecoin-project/go-filecoin/issues/2210)

**Comment by shannonwells** [comment]
Ok thanks!

**Comment by acruikshank** [comment]
This problem is slightly different than #2210. It looks like a failure to connect to the miner rather than a problem trying to create the payment channel. At the time this issue was created this was working as expected (storage deals are pretty rough right now). Since then we've added #2216 to check that the connection is open before creating the payments. This should make communication failures less expensive.

I'm going to close for now as this problem is somewhat addressed with #2216, and other types failures are covered in #2210.
